x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -147,6 +177,18 @@ void addQueryFailure(int shardIndex, ShardSearchFailure shardSearchFailure) {
     }
 
     private SearchResponse buildResponse(long taskStartTimeNanos, InternalAggregations reducedAggs) {
+        if (clustersInfo.getClusters() != null && clustersInfo.isLocalClusterStatusUpdated() == false) {
+            synchronized (this) {
```

Comment:
Does this section need to be thread safe? Can `buildResponse` be called by two threads on the same `MutableSearchReponse` object? (Since other methods in this class are `synchronized`, I'm guessing the answer is yes, but want to double check with the reviewers.)

Reply:
yes, but buildResponse is only called from synchronized methods, so no additional synchronization is needed?